_dev/js/front/src/components/common/card-fields.component.js

```diff
@@ -0,0 +1,322 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ */
+
+/**
+ * @typedef PaypalCardFieldCardField
+ * @type {*}
+ *
+ * @property {boolean} isEmpty
+ * @property {boolean} isValid
+ * @property {boolean} isPotentiallyValid
+ * @property {boolean} isFocused
+ */
+
+/**
+ * @typedef PaypalCardFieldsEvent
+ * @type {*}
+ *
+ * @property {string} emittedBy
+ * @property {boolean} isFormValid
+ * @property {String[]} errors
+ * @property {*} fields
+ * @property {PaypalCardFieldCardField} fields.cardCvvField
+ * @property {PaypalCardFieldCardField} fields.cardExpiryField
+ * @property {PaypalCardFieldCardField} fields.cardNameField
+ * @property {PaypalCardFieldCardField} fields.cardNumberField
+ */
+
+import { BaseComponent } from '../../core/dependency-injection/base.component';
+
+export class CardFieldsComponent extends BaseComponent {
+  static Inject = {
+    config: 'PsCheckoutConfig',
+    configPayPal: 'PayPalSdkConfig',
+    payPalService: 'PayPalService',
+    psCheckoutApi: 'PsCheckoutApi',
+    psCheckoutService: 'PsCheckoutService',
+    querySelectorService: 'QuerySelectorService',
+  };
+
+  created() {
+    this.data.name = this.props.fundingSource.name;
+    this.data.validity = false;
+    /**
+     * @property {PaypalCardFieldsEvent} data.cardFieldsState
+     */
+    this.data.cardFieldsState = {};
+
+    this.data.cardFieldsFocused = {
+      name: false,
+      number: false,
+      expiry: false,
+      cvv: false,
+    }
+    this.data.HTMLElement = this.props.HTMLElement;
+    this.data.HTMLElementCardForm = this.querySelectorService.getCardFieldsFormContainer();
+    this.data.HTMLElementBaseButton = this.querySelectorService.getBasePaymentConfirmation();
+    this.data.HTMLElementButton = null;
+    this.data.HTMLElementButtonWrapper = this.getButtonWrapper();
+
+    this.data.HTMLElementCardHolderName = this.querySelectorService.getCardFieldsNameInputContainer();
+    this.data.HTMLElementCardNumber = this.querySelectorService.getCardFieldsNumberInputContainer();
+    this.data.HTMLElementCardExpiry = this.querySelectorService.getCardFieldsExpiryInputContainer();
+    this.data.HTMLElementCardCvv = this.querySelectorService.getCardFieldsCvvInputContainer();
+
+    this.data.HTMLElementCardNameError= this.querySelectorService.getCardFieldsNameError();
+    this.data.HTMLElementCardNumberError= this.querySelectorService.getCardFieldsNumberError();
+    this.data.HTMLElementCardVendorError= this.querySelectorService.getCardFieldsVendorError();
+    this.data.HTMLElementCardExpiryError= this.querySelectorService.getCardFieldsExpiryError();
+    this.data.HTMLElementCardCvvError= this.querySelectorService.getCardFieldsCvvError();
+  }
+
+  getButtonWrapper() {
```

Comment:
@Matt75 This was the only selector that I couldn't move to querySelectorService because it uses variable selector